src/GLFW.jl

```diff
@@ -10,38 +10,45 @@ function GetVersion()
 	ccall( (:glfwGetVersion, lib), Cvoid, (Ref{Cint}, Ref{Cint}, Ref{Cint}), major, minor, rev)
 	VersionNumber(major[], minor[], rev[])
 end
-const libversion = GetVersion()
-
-if libversion.major == 3
-	include("callback.jl")
-	include("glfw3.jl")
-	include("vulkan.jl")
-	if isdefined(Base, :getproperty) && isdefined(Base, :setproperty!)
-		# Julia 0.7 supports property overloading
-		# TODO for Julia upgrade: remove feature guard
-		include("monitor_properties.jl")
-	end
-else
-	error("GLFW $libversion is not supported")
+
+include("callback.jl")
+include("glfw3.jl")
+include("vulkan.jl")
+
+if isdefined(Base, :getproperty) && isdefined(Base, :setproperty!)
+	# Julia 0.7 supports property overloading
+	# TODO for Julia upgrade: remove feature guard
+	include("monitor_properties.jl")
 end
 
+const GLFW_INITIALIZED = Ref(false)
+
+is_initialized() = GLFW_INITIALIZED[]
+
+
 function __init__()
-	initialized = false
+	libversion = GetVersion()
 
+	if libversion.major != 3
+		error("GLFW version $libversion not supported")
+	end
 	# Save errors that occur during initialization
 	errors = Vector{Exception}()
 	SetErrorCallback(err -> push!(errors, err))
 
 	try
-		initialized = GLFW.Init()
+		GLFW_INITIALIZED[] = GLFW.Init()
```

Comment:
I suggest moving writes to `GLFW_INITIALIZED` inside the `Init` and `Destroy` wrappers. Then it will always be accurate regardless of where the calls are made.